app/src/main/java/com/github/geohunt/app/maps/marker/DisplayMarkers.kt

```diff
@@ -61,55 +80,56 @@ fun MarkerInfoWindowContent(
         Column(
             horizontalAlignment = Alignment.CenterHorizontally
         ) {
+            Spacer(modifier = Modifier.height(16.dp))
+
             //The image displayed at the top of the info window
             if (challenge.image != "") {
                 Column(
                     modifier = Modifier
                         .fillMaxWidth()
                         .testTag("Marker image")
                 ) {
-                    Image(
-                        painter = rememberAsyncImagePainter(
-                            ImageRequest.Builder(LocalContext.current).data(data = challenge.image)
-                                .apply(block = fun ImageRequest.Builder.() {
-                                    crossfade(true)
-                                    allowHardware(false)
-                                    scale(Scale.FILL)
-                                }).build()
-                        ),
-                        contentDescription = "Marker Image",
-                        contentScale = ContentScale.Fit,
+                    val imagePainter = ImageRequest.Builder(LocalContext.current)
+                            .data(data = challenge.image)
+                            .apply(block = fun ImageRequest.Builder.() {
+                                crossfade(true)
+                                allowHardware(false)
+                                scale(Scale.FILL)
+                                listener(
+                                    onSuccess = { _, _ ->
+                                        imageLoaded.value = true
+                                    }
+                                )
+                            }).build()
+
+                    AsyncImage(
+                        model = imagePainter,
                         modifier = Modifier
-                            .size(70.dp)
+                            .size(160.dp)
                             .align(Alignment.CenterHorizontally)
+                            .testTag("Marker image"),
+                        contentDescription = if (imageLoaded.value) "Marker Image" else "Loading",
                     )
```

Comment:
If I understand your comment correctly, you're suggesting to replace
```
                    val imagePainter = ImageRequest.Builder(LocalContext.current)
                            .data(data = challenge.image)
                            .apply(block = fun ImageRequest.Builder.() {
                                crossfade(true)
                                allowHardware(false)
                                scale(Scale.FILL)
                                listener(
                                    onSuccess = { _, _ ->
                                        imageLoaded.value = true
                                    }
                                )
                            }).build()

                    AsyncImage(
                        model = imagePainter,
                        modifier = Modifier
                            .size(160.dp)
                            .align(Alignment.CenterHorizontally)
                            .testTag("Marker image"),
                        contentDescription = if (imageLoaded.value) "Marker Image" else "Loading",
                    )
```

by:

```
                    AsyncImage(
                        model = imagePainter,
                        modifier = Modifier
                            .size(160.dp)
                            .align(Alignment.CenterHorizontally)
                            .testTag("Marker image"),
                        contentDescription = if (imageLoaded.value) "Marker Image" else "Loading",
                    )
```

This would lead to the error:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.github.geohunt.app, PID: 7693
    java.lang.IllegalArgumentException: Software rendering doesn't support hardware bitmaps
        at android.graphics.BaseCanvas.throwIfHwBitmapInSwMode(BaseCanvas.java:706)
        at android.graphics.BaseCanvas.throwIfCannotDraw(BaseCanvas.java:81)
        at android.graphics.BaseCanvas.drawBitmap(BaseCanvas.java:139)
```
        ...

Because you do not enforce `allowHardware(false)`, that's why I use the ImageRequest.Builder

Reply:
Ok, sorry. What I meant was that you can replace all of the logic "when the image is being loaded" by the placeholder. I think it is either a method of the builder, either an argument that can be passed directly to ImageAsync. This would simplify the logic quite a bit

Reply:
I have simplified the logic of the AsyncImage